packages/syft/src/syft/core/node/new/task/task_service.py

```diff
@@ -0,0 +1,400 @@
+                )
+            enclave_task.status[task_owner] = task.status[task_owner]
+            # Create nested action store of DO for storing intermediate data
+            if not self.action_store.exists(task_owner.node_uid):
+                dict_object = DictObject()
+                dict_object.base_dict[task.id] = private_input_map
+                self.action_store.set(
+                    uid=task_owner.node_uid,
+                    credentials=context.credentials,
+                    syft_object=dict_object,
+                )
+            else:
+                result = self.action_store.get(
+                    uid=task_owner.node_uid, credentials=context.credentials
+                )
+                if result.is_ok():
+                    result = result.ok()
+                    result.base_dict[task.id] = private_input_map
+                else:
+                    return result.err()
+
+            # Check if all the DO approve the code execution
+            code_status_check = set(enclave_task.status.values())
+            if len(code_status_check) == 1 and "Approved" in code_status_check:
+                enclave_task.execution = "executing"
+                self.task_stash.update(enclave_task)
+                # TODO 🟣 Branch to separate thread for execution of enclave task
+                self.execute_task(enclave_task=enclave_task, context=context)
+
+            self.task_stash.update(enclave_task)
+
+            return Ok(True)
+        else:
+            return Err(
+                f"Task: {task.id.no_dash}:{enclave_task.owners} in enclave does not contain {task.owners[0].name} "
+            )
+
+    def fetch_private_inputs(
+        self, enclave_task: Task, context: AuthedServiceContext
+    ) -> Dict:
+        inputs = {}
+        for domain in enclave_task.owners:
+            # TODO 🟣 Sometimes the pydantic mapping is converted to dict values
+            domain = NodeView(**domain) if isinstance(domain, dict) else domain
+            domain_input = self.action_store.get(
+                uid=domain.node_uid,
+                credentials=context.credentials,
+                skip_permission=True,
+            )
+            if domain_input.is_ok():
+                inputs.update(domain_input.ok().base_dict[enclave_task.id])
+            else:
+                return domain_input.err()
+        return inputs
+
+    def execute_task(self, enclave_task: Task, context: AuthedServiceContext) -> None:
+        global stdout_
+        global stderr_
+
+        code = enclave_task.code
+        inputs = self.fetch_private_inputs(enclave_task=enclave_task, context=context)
+        outputs = enclave_task.outputs
+
+        try:
+            logger.info(f"inital outputs: {outputs}")
+
+            # Check overlap between inputs and vars
+            global_input_inter = set(globals().keys()).intersection(set(inputs.keys()))
```

Comment:
Q: I don't think you need to use `set` on the `inputs.key()`/`vars.keys()` . I think it should already be a set.
(also for the next line).